fdbserver/DDRelocationQueue.actor.cpp

```diff
@@ -530,6 +530,35 @@ struct DDQueue : public IDDRelocationQueue {
 		Future<Void> cancel;
 	};
 
+	struct MovingAverage {
+		int64_t previous;
+		int64_t total;
+		double interval;
+		Deque<std::pair<double, int64_t>> updates; // std::pair<time, bytes>
+
+		int64_t getTotal() { return total; }
+
+		double getAverage() {
+			while (!updates.empty() && updates.front().first < now() - interval) {
+				previous += updates.front().second;
+				updates.pop_front();
+			}
+			return (total - previous) / interval;
```

Comment:
This will return an incorrect value if `getAverage()` is called less than `interval` time after the struct is created. This corner case can be fixed in a future PR if you'd like, but then it's good to call it out in a comment. The fix is probably pretty simple. One idea would be to push a 0 byte update when the struct is created. Along with the change suggested below for preventing OOMs (i.e., dividing by timestamp diff rather than interval), this should return the correct value.